src/components/GenericWarningModal/GenericWarningModal.tsx

```diff
@@ -0,0 +1,46 @@
+import {
+  Text,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react"
+import { ModalCloseButton } from "@opengovsg/design-system-react"
+import parse from "html-react-parser"
+
+interface ModalProps {
+  isOpen: boolean
+  onClose: () => void
+  displayTitle: string
+  displayText: string
+  children: JSX.Element | JSX.Element[]
```

Comment:
prefer using `PropsWithChildren` instead of manual typing as the types themselves don't guarantee that all possible jsx elements that can be children are children (eg: differing typing in `ReactNode` vs `JSX.Element`. [this](https://www.carlrippon.com/react-children-with-typescript/) article provides a brief overview of the many available types but because it's so confusing, the type that react provides should almost always be preferred

Reply:
Makes sense, changed in https://github.com/isomerpages/isomercms-frontend/pull/941/commits/eb973f7885ef6e088e30376fd92c04cfc156f085